shared/common-adapters/name-with-icon.js

```diff
@@ -0,0 +1,192 @@
+// @flow
+import * as React from 'react'
+import Avatar from './avatar'
+import Box from './box'
+import ClickableBox from './clickable-box'
+import Icon from './icon'
+import {type IconType} from './icon.constants'
+import Text, {type TextType} from './text'
+import {Usernames} from './usernames'
+import {collapseStyles, globalStyles, isMobile, styleSheetCreate} from '../styles'
+
+type Size = 'small' | 'default' | 'large'
+
+// Exposed style props for the top-level container and box around metadata arbitrarily
+type Props = {
+  following?: boolean,
```

Comment:
lets not plumb through isYou/following/etc. Lets just use ConnectedUsername always